packages/suite/src/utils/suite/router.ts

```diff
@@ -68,15 +68,22 @@ const validateWalletParams = (url: string): CommonWalletParams => {
     };
 };
 
-const validateModalAppParams = (url: string) => {
+const parseParamValue = (value: string) => {
+    if (value === 'true') return true;
+    if (value === 'false') return false;
+
+    return value;
+};
+
+const validateModalAppParams = (url: string, params?: Route['params']) => {
     const [, hash] = stripPrefixedURL(url).split('#');
     if (!hash) return;
-    const [cancelable] = hash.split('/').filter(p => p.length > 0);
-    if (cancelable !== 'true') return;
+    const splitted = hash.split('/').filter(p => p.length > 0);
+    if (splitted.length === 0) return;
```

Comment:
L85 can access out of bounds index from `splitted`, returning undefined, and idk how that will behave..
This will handle it gracefully I think
```suggestion
    if (params === undefined) return;
    if (splitted.length < params?.length) return;
```

Reply:
cancelable can be there